artview/core/common.py

```diff
@@ -155,42 +155,49 @@ def startDisplay(self):
         return self.result, self.independent.isChecked()
 
 ########################
-# Table methods #
+#    Table methods     #
 ########################
 
 
 class CreateTable(QtGui.QTableWidget):
     """Creates a custom table widget."""
-    def __init__(self, column_names, name="Table",
+    def __init__(self, points, name="Table",
                  textcolor="black", bgcolor="gray", parent=None, *args):
         QtGui.QTableWidget.__init__(self, *args)
+        self.points = points
         self.setSelectionMode(self.ContiguousSelection)
         self.setGeometry(0, 0, 700, 400)
         self.setShowGrid(True)
         self.textcolor = textcolor
         self.bgcolor = bgcolor
 
-        self.colnames = column_names
-
-    def display_data(self, data):
+    def display(self):
         """Reads in data from a 2D array and formats and displays it in
             the table."""
 
-        if len(data) == 0:
+        if self.points is None:
             data = ["No data for selected."]
             nrows = 0
             ncols = 0
         else:
-            nrows, ncols = data.shape[0], data.shape[1]
+            nrows = self.points.npoints
+            colnames = self.points.axes.keys() + self.points.fields.keys()
+            ncols = len(colnames)
 
         self.setRowCount(nrows)
         self.setColumnCount(ncols)
-        self.setHorizontalHeaderLabels(self.colnames)
+        colnames = self.points.axes.keys() + self.points.fields.keys()
+        self.setHorizontalHeaderLabels(colnames)
 
         for i in xrange(nrows):
             # Set each cell to be a QTableWidgetItem from _process_row method
-            for j in xrange(ncols):
-                item = QtGui.QTableWidgetItem(str(data[i, j]).format("%8.3f"))
+            for j, name in enumerate(colnames):
+                if name in self.points.axes:
+                    item = QtGui.QTableWidgetItem(str(
+                        self.points.axes[name]['data'][i]).format("%8.3f"))
```

Comment:
I'm not sure this format code is being applied to x_axis and y_axis, when I put up table, there are at least 6 decimal digits. The others display fine, any idea why this is?